src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/ApplyVQSR.java

```diff
@@ -0,0 +1,628 @@
+        vcfWriter = GATKVariantContextUtils.createVCFWriter(
+                new File(output),
+                getBestAvailableSequenceDictionary(),
+                lenientVCFProcessing);
+        vcfWriter.writeHeader(vcfHeader);
+    }
+
+    private boolean trancheIntervalIsValid(final String sensitivityLimits) {
+        final String[] vals = sensitivityLimits.split("to");
+        if(vals.length != 2)
+            return false;
+        try {
+            double lowerLimit = Double.parseDouble(vals[0]);
+            double upperLimit = Double.parseDouble(vals[1].replace("+",""));    //why does our last tranche end with 100+? Is there anything greater than 100 percent?  Really???
+        }
+        catch(NumberFormatException e) {
+            throw new UserException("Poorly formatted tranche filter name does not contain two sensitivity interval end points.");
+        }
+        return true;
+    }
+
+    /**
+     * Check the filter declarations in the input VCF header to see if any ApplyRecalibration mode has been run
+     * Here we assume that the tranches are named with a specific format: VQSRTranche[SNP|INDEL][lowerLimit]to[upperLimit]
+     * @param inputHeaders
+     */
+    private void checkForPreviousApplyRecalRun(final Set<VCFHeaderLine> inputHeaders) {
+        for(final VCFHeaderLine header : inputHeaders) {
+            if(header instanceof VCFFilterHeaderLine) {
+                final String filterName = ((VCFFilterHeaderLine)header).getID();
+                //TODO: clean up these magic numbers
+                if(filterName.length() < 12 || !filterName.substring(0, 11).equalsIgnoreCase(trancheFilterString)) {
+                    continue;
+                }
+                if(filterName.charAt(11) == 'S') {
+                    //for SNP tranches, get sensitivity limit
+                    final String sensitivityLimits = filterName.substring(14);
+                    if(trancheIntervalIsValid(sensitivityLimits))
+                        foundSNPTranches = true;
+                }
+                else if(filterName.charAt(11) == 'I') {
+                    //for INDEL tranches, get sensitivity limit
+                    final String sensitivityLimits = filterName.substring(16);
+                    if(trancheIntervalIsValid(sensitivityLimits))
+                        foundINDELTranches = true;
+                }
+            }
+        }
+    }
+
+    //---------------------------------------------------------------------------------------------------------------
+    //
+    // map
+    //
+    //---------------------------------------------------------------------------------------------------------------
+
+    @Override
+    public void apply(final VariantContext vc, final ReadsContext readsContext, final ReferenceContext ref, final FeatureContext featureContext) {
+
+        final List<VariantContext> VCs =  featureContext.getValues(getDrivingVariantsFeatureInput(), featureContext .getInterval().getStart());
```

Comment:
Also, I think with the existing implementation you'd end up processing the same variants multiple times, if there are multiple variants sharing the same start position, which is explicitly allowed in the VCF spec. Can you add a test with multiple variants with the same start? (although the best solution is just to find a way to use the `vc` passed in directly and get rid of the extra query)